AzureMonitorAgent/ama_tst/modules/connect/connect.py

```diff
@@ -0,0 +1,140 @@
+                        dcr_region.add(region)
+                        # parse url suffix
+                        if '.us' in endpoint_url:
+                            general_info['URL_SUFFIX'] = '.us'
+                        if '.cn' in endpoint_url:
+                            general_info['URL_SUFFIX'] = '.cn'                            
+                    if channel['protocol'] == 'me':
+                        # parse ME region
+                        endpoint_url = channel['endpoint']
+                        region = endpoint_url.split('https://')[1].split('.monitoring')[0]
+                        me_region.add(region)
+    except (FileNotFoundError, AttributeError) as e:
+        error_info.append((e,))
+        return ERR_NO_DCR
+
+    general_info['DCR_WORKSPACE_ID'] = dcr_workspace
+    general_info['DCR_REGION'] = dcr_region
+    general_info['ME_REGION'] = me_region
+    return NO_ERROR
+
+def check_subcomponents(): 
+    services = ['azuremonitoragent', 'metrics-sourcer', 'azuremonitor-agentlauncher',
+                     'azuremonitor-coreagent','metrics-extension']
+    for service in services:
+        try:
+            status = subprocess.check_output(['systemctl', 'status', service],\
+                                    universal_newlines=True, stderr=subprocess.STDOUT)
+            status_lines = status.split('\n')
+            for line in status_lines:
+                line = line.strip()
+                if line.startswith('Active:'):
+                    if not line.split()[1] == 'active':
+                        error_info.append((service, status))
+                        return ERR_SUBCOMPONENT_STATUS
+        except subprocess.CalledProcessError as e:
+            error_info.append((e,))
+            return ERR_CHECK_STATUS
+            
+    return NO_ERROR
+
+def check_connection(interactive, err_codes=True, prev_success=NO_ERROR):
+    print("CHECKING CONNECTION...")
+
+    success = prev_success
+    
+    # check /etc/default/azuremonitoragent file
+    print("Checking AMA parameters in /etc/default/azuremonitoragent...")
+    checked_parameters = check_parameters()
+    if (is_error(checked_parameters)):
+        return print_errors(checked_parameters)
+    else:
+        success = print_errors(checked_parameters)
+        
+    # check DCR
+    print("Checking DCR...")
+    checked_workspace = check_workspace()
+    if (is_error(checked_workspace)):
+        return print_errors(checked_workspace)
+    else:
+        success = print_errors(checked_workspace)
+
+    # check general internet connectivity
+    print("Checking if machine is connected to the internet...")
+    checked_internet_connect = check_internet_connect()
+    if (is_error(checked_internet_connect)):
+        return print_errors(checked_internet_connect)
+    else:
+        success = print_errors(checked_internet_connect)
+
+
+    # check if AMA endpoints connected
+    print("Checking if AMA endpoints are connected...")
```

Comment:
nit on wording, IMO would be clearer to say "Checking if machine can connect to Azure Monitor control-plane and data ingestion endpoints..."